apps/yggdrasil/lib/yggdrasil/role.ex

```diff
@@ -0,0 +1,10 @@
+defmodule Yggdrasil.Role do
+  use Ecto.Schema
+
+  schema "roles" do
+    field :name, :string
+    has_many :role_resources, Yggdrasil.RoleResource
```

Comment:
I was about to say something similar to `users_roles` below, but this is an odd case. This isn't simply a association table `users_tables`, because it adds in resource, so it's not technically a permission. So it's I think slightly confusing to call it `roles.permissions`. 

```Elixir
Enum.map role.permissions, fn p ->
  # seems more awkward for now
  # I would expect the p to the be the permission
  p.permission
end
```